### ISSUE TITLE
It is not possible to lint module with objectpath library

### ISSUE THREAD

**Issue by pchomik** [PyCQA/pylint#2652]
### Steps to reproduce
1. Create file like below

```
from objectpath import Tree


class ElasticDataRequest(object):

    def __init__(self):
        self.data = {
            'must': {
                'range': {
                    '@timestamp': {
                        'gte': 0,
                        'lte': 0
                    }
                }
            }
        }

    def set_time_range(self, start=None, end=None):
        tree = Tree(self.data)
        for element in tree.execute('$..must.*.range'):
            if start:
                element['@timestamp']['gte'] = start
            if end:
                element['@timestamp']['lte'] = end
        return self
```

2. Install objectpath, pylint on Python 3.6.2
3.Run command pylint module_name.py

### Current behavior

Application freezes after few lines of output
```
************* Module error_test
error_test.py:7:0: C0111: Missing class docstring (missing-docstring)
error_test.py:7:0: R0205: Class 'ElasticDataRequest' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
error_test.py:21:4: C0111: Missing method docstring (missing-docstring)
```

### Expected behavior

Module can be linted without freeze.

### pylint --version output

I will share all installed libraries in virtualenv:

astroid (2.1.0)
isort (4.3.4)
lazy-object-proxy (1.3.1)
mccabe (0.6.1)
objectpath (0.6.1)
pip (9.0.1)
pylint (2.2.2)
setuptools (28.8.0)
six (1.12.0)
typed-ast (1.1.1)
wrapt (1.10.11)

### Additional information

pylint freezes only if Tree instance is created.

**Comment by pchomik** [comment]
Hi @PCManticore 

I installed astroid version from mentioned review and linter is working - at least is able to finish.
Thanks for support.

**Comment by DanielNoord** [comment]
I think this was fixed by the fix that closed #2198.

I get a normal running time for this on `main`, therefore I'm closing this. Let me know if you're still experiencing issues.
